app/models/app.rb

```diff
@@ -58,6 +58,34 @@ def maintenance
     redis.get(maintenance_key)
   end
 
+  def whitelist
+    redis.smembers(whitelist_key) || []
+  end
+
+  def whitelist=(domains)
+    whitelist.each { |domain| remove_domain(domain) }
```

Comment:
I had the same thought, but since dependency was doing the same thing, figured you might want an audit each time. Redis also has intersection and difference commands we could use.